### PR TITLE
 NEW RECORD TYPE: HTTPS & SVCB 

### PR DESCRIPTION
**Pull request by fritterhoff**

### CODE REVIEW

**Comment by fritterhoff** [comment]
That should fix #1660 if I complete all required additions and tests... :)

**Comment by fritterhoff** [comment]
@tlimoncelli I'm a little bit unhappy about the SVCB params parsing. Any advice/hint how I can improve that and avoid the code duplication?

**Comment by tlimoncelli** [comment]
> @tlimoncelli I'm a little bit unhappy about the SVCB params parsing. Any advice/hint how I can improve that and avoid the code duplication?

How about letting miekg/dns do the parsing like in SetTargetLOCString() ?

**Comment by fritterhoff** [comment]
Thanks for the hint. I already changed it!

**Comment by tlimoncelli** [comment]
Looks good so far!  When the PR is no longer marked draft, I'll do a deeper review.

One note: Please add integration tests that (1) create a record, (2) change each field one at a time.

Thanks!

**Comment by fritterhoff** [comment]
Will hopefully complete the changes tomorrow and extend also the docs.

**Comment by fritterhoff** [comment]
Prop. I missed some places (again) ... from my point of view the most changes should be done.

File: build/generate/featureMatrix.go
Comment (cafferata) [inline build/generate/featureMatrix.go:83]:
```suggestion
		DomainModifierHttps  = "[`HTTPS`](language-reference/domain-modifiers/HTTPS.md)"
```

File: build/generate/featureMatrix.go
Comment (cafferata) [inline build/generate/featureMatrix.go:90]:
```suggestion
		DomainModifierSvcb   = "[`SVCB`](language-reference/domain-modifiers/SVCB.md)"
```

File: documentation/language-reference/domain-modifiers/HTTPS.md
Comment (cafferata) [inline documentation/language-reference/domain-modifiers/HTTPS.md:2]: This page needs to be added to `SUMMARY.md`, see [How to add a new page?](https://docs.dnscontrol.org/developer-info/styleguide-doc#how-to-add-a-new-page).

File: documentation/language-reference/domain-modifiers/SVCB.md
Comment (cafferata) [inline documentation/language-reference/domain-modifiers/SVCB.md:2]: This page needs to be added to `SUMMARY.md`, see [How to add a new page?](https://docs.dnscontrol.org/developer-info/styleguide-doc#how-to-add-a-new-page).

**Review by cafferata — CHANGES_REQUESTED** [review]
This GitHub pull request looks great! I only have some minor feedback. ☺️

File: providers/bind/bindProvider.go
Comment (cafferata) [inline providers/bind/bindProvider.go:50]: @tlimoncelli, would it be an idea to call on all provider maintainers and ask them if they would like to test these two capabilities and determine whether these capabilities can be used for their provider?

Comment (cafferata) [reply providers/bind/bindProvider.go:50]: _Obviously outside the scope of this GitHub pull request._

Comment (fritterhoff) [reply providers/bind/bindProvider.go:50]: I can at least confirm it is working also for Cloudflare.

**Comment by fritterhoff** [comment]
> This GitHub pull request looks great! I only have some minor feedback. ☺️

Thanks for the feedback. Fixed the issues in the docs.

**Review by cafferata — APPROVED** [review]

File: documentation/language-reference/domain-modifiers/HTTPS.md
Comment (tlimoncelli) [inline documentation/language-reference/domain-modifiers/HTTPS.md:17]:
```suggestion
HTTPS adds an HTTPS record to a domain. The name should be the relative label for the record. Use `@` for the domain apex. The HTTPS record is a special form of the SVCB resource record.
```

File: documentation/language-reference/domain-modifiers/SVCB.md
Comment (tlimoncelli) [inline documentation/language-reference/domain-modifiers/SVCB.md:17]:
```suggestion
SVCB adds an SVCB record to a domain. The name should be the relative label for the record. Use `@` for the domain apex.
```

File: integrationTest/integration_test.go
Comment (tlimoncelli) [inline integrationTest/integration_test.go:1037]: Only change one parameter at a time.

NOTE: I'm not an expert in the HTTPS/SVCB standard. Please verify if these are realistic parameters. I could be way off base!

```suggestion
			tc("Create a HTTPS record", https("@", 1, "test.com.", "port=80")),
			tc("Change HTTPS priority", https("@", 2, "test.com.", "port=80")),
			tc("Change HTTPS target", https("@", 2, ".", "port=80")),
			tc("Change HTTPS params", https("@", 2, ".", "port=99")),
			tc("Change HTTPS params-empty", https("@", 2, ".", "")),
			tc("Change HTTPS all", https("@", 3, "example.com.", "port=100")),
```

File: integrationTest/integration_test.go
Comment (tlimoncelli) [inline integrationTest/integration_test.go:1044]:
```suggestion

```suggestion
			tc("Create a SVCB record", svcb("@", 1, "test.com.", "port=80")),
			tc("Change SVCB prority", svcb("@", 2, "test.com.", "port=80")),
			tc("Change SVCB target", svcb("@", 2, ".", "port=80")),
			tc("Change SVCB params", svcb("@", 2, ".", "port=99")),
			tc("Change SVCB params-empty", svcb("@", 2, ".", "")),
			tc("Change SVCB all", svcb("@", 3, "example.com.", "port=100")),
``````

File: models/t_svcb.go
Comment (tlimoncelli) [inline models/t_svcb.go:40]:
```suggestion
	case *dns.HTTPS, *dns.SVCB:
		return rc.SetTargetSVCB(r.Priority, r.Target, r.Value)
```

File: providers/bind/bindProvider.go
Comment (tlimoncelli) [reply providers/bind/bindProvider.go:50]: Yes, once this is merged please post to the mailing list and ask maintainers to add support.

**Review by tlimoncelli — COMMENTED** [review]
Mostly just some cosmetic changes.

One more thing... please reformat helpers.js:

```
npm install prettier
node_modules/.bin/prettier --write pkg/js/helpers.js
```

(Please do NOT check in the new `package-lock.json` file)

File: models/t_svcb.go
Comment (fritterhoff) [reply models/t_svcb.go:40]: That change sadly does not work :(

Comment (tlimoncelli) [reply models/t_svcb.go:41]: Ok!

**Pull request merged**
